src/main/java/io/quarkus/gizmo/Gizmo.java

```diff
@@ -1110,13 +1113,13 @@ private void generateEquals() {
                                 .falseBranch().returnBoolean(false);
                         break;
                     case "F": // float
-                        // could use `Float.compare` instead of `==`
-                        equals.ifZero(equals.compareFloat(thisValue, thatValue, false))
+                        // this is consistent with Arrays.equals() and it's also what IntelliJ generates
+                        equals.ifIntegerEqual(equals.invokeStaticMethod(FLOAT_TO_INT_BITS, thisValue), equals.invokeStaticMethod(FLOAT_TO_INT_BITS, thatValue))
```

Comment:
It's pretty wild that `Arrays.equals()` is doing bitwise equality. 🤯

Importantly, this also makes this method consistent with `Float.hashCode()` which is doing `floatToIntBits` as well.

Reply:
Ha, I didn't even notice what `Float|Double.hashCode()` does, good point.